framework/sandstone.cpp

```diff
@@ -1553,7 +1553,7 @@ static void wait_for_children(ChildrenList &children, int *tc, const struct test
     children.pollfds.emplace_back(pollfd{ .fd = sApp->shmem->server_debug_socket, .events = POLLIN });
     auto remove_debug_socket = scopeExit([&] { children.pollfds.pop_back(); });
 
-    auto single_wait = [&](milliseconds timeout) {
+    auto single_wait = [&, caughtSignal = 0](milliseconds timeout) mutable {
```

Comment:
simple conditions (timeout '==0', signal '<0' and event otherwise, with clear returns) would be simpler to read/understand.